sonar/modules/projects/query.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# Swiss Open Access Repository
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Query for projects."""
+
+from flask import current_app, request
+
+from sonar.modules.organisations.api import current_organisation
+from sonar.modules.query import default_search_factory
+from sonar.modules.users.api import current_user_record
+
+
+def search_factory(self, search, query_parser=None):
+    """Project search factory.
```

Comment:
Can you explain how it is different from the default factory?

Reply:
Because it does additional checks with the user context.